crates/apollo-router/src/router_factory.rs

```diff
@@ -44,4 +62,37 @@ impl RouterFactory<ApolloRouter, ApolloPreparedQuery> for ApolloRouterFactory {
         .map(|res| res.expect("ApolloRouter::new() is infallible; qed"))
         .boxed()
     }
+
+    fn recreate(
+        &self,
+        graph: Arc<ApolloRouter>,
+        configuration: &Configuration,
+        schema: Arc<graphql::Schema>,
+        plan_cache_limit: usize,
+    ) -> future::BoxFuture<'static, ApolloRouter> {
+        let factory = self.create(configuration, schema, plan_cache_limit);
+
+        tokio::task::spawn(async move {
+            // Use the "hot" entries in the supplied graph to pre-populate
+            // our new graph
+            let new_graph = factory.await;
+            let hot_keys = graph.get_query_planner().get_hot_keys().await;
+            // It would be nice to get these keys concurrently by spawning
+            // futures in our loop. However, these calls to get call the
+            // v8 based query planner and running too many of these
+            // concurrently is a bad idea. One for the future...
+            for key in hot_keys {
+                // We can ignore errors, since we are just warming up the
+                // cache
+                let _ = new_graph.get_query_planner().get(key.0, key.1, key.2).await;
+            }
+            new_graph
+        })
+        .map(|res| res.expect("recreate() is infallible; qed"))
+        .boxed()
```

Comment:
`recreate` actually is async because it returns a BoxFuture. The `async` syntax of Rust is just sugar for: `impl Future<Output = RESULT> + Send>` (more or less).


```suggestion
        Box::pin(async move {
            // Use the "hot" entries in the supplied graph to pre-populate
            // our new graph
            let new_graph = factory.await;
            let hot_keys = graph.get_query_planner().get_hot_keys().await;
            // It would be nice to get these keys concurrently by spawning
            // futures in our loop. However, these calls to get call the
            // v8 based query planner and running too many of these
            // concurrently is a bad idea. One for the future...
            for key in hot_keys {
                // We can ignore errors, since we are just warming up the
                // cache
                let _ = new_graph.get_query_planner().get(key.0, key.1, key.2).await;
            }
            new_graph
        })
```

Reply:
Ah, thank you! Done.